pkg/validation/providers.go

```diff
@@ -96,3 +123,20 @@ func validateGoogleConfig(provider options.Provider) []string {
 
 	return msgs
 }
+
+func validateEntraConfig(provider options.Provider) []string {
+	msgs := []string{}
+
+	if provider.MicrosoftEntraIDConfig.FederatedTokenAuthEnabled {
+		federatedTokenPath := os.Getenv("AZURE_FEDERATED_TOKEN_FILE")
```

Comment:
Where is this "magic" file and environment variable coming from?

Shouldn't this be a flag? Why is the token supposed to be inside a file instead of directly inside the env var?

Reply:
This is how Workload Identity works, token path is injected as an env var:
https://azure.github.io/kubelogin/concepts/login-modes/workloadidentity.html

Reply:
Okay got it. So the `magic` is happening because of a controller managed through:

> * Kubernetes service account associated with oauth2-proxy deployment, is annotated with `azure.workload.identity/client-id: <app-registration-client-id>`
> * oauth2-proxy pod is labeled with `azure.workload.identity/use: "true"`